#167164985 View User Profile

allebd opened the pull request:
#### Pivotal tracker story
[#167164985](https://www.pivotaltracker.com/story/show/167164985)
#### What does this PR do?
View user profile
#### Summary of Task
- write using test to view user profile by userId
- create a route to view profile
- create user controller to view profile
#### How can this be tested?
- Use **git clone** to clone this repository to local branch
- Use **npm install** to download all necessary packages
- Ensure your database is created or you can use **sequelize db:create** if database does not exist
- Use **npm run test**  to test
- Go your POSTMAN and type **http://localhost:${PORT}/api/v1/profiles/:userId** to view a particular user by userId
#### Screenshots (if appropriate)
<img width="675" alt="Screenshot 2019-08-02 at 3 49 02 PM" src="https://user-images.githubusercontent.com/7049701/62378950-cee11200-b53d-11e9-875b-31b869faf454.png">
Screenshot of a successful view of user profile

<img width="702" alt="Screenshot 2019-08-02 at 3 55 27 PM" src="https://user-images.githubusercontent.com/7049701/62379022-fb952980-b53d-11e9-92ad-8d1dd8892918.png">
Screenshot of a user profile not found

OvieMudi reviewed (approved):
LGTM

Ukhu commented on src/routes/user.js:
```diff
@@ -1,11 +1,15 @@
 
-user.post('/users', middlewares, signUp);
+// Route to get user profiles by username
+user.get('/profiles/:username', profile);
```
I noticed there is no validation middleware on this route to ensure that the username entered isn't empty and is valid.

allebd replied:
Noted, would work on it

allebd replied:
Done

benisi reviewed (approved):
LGTM

Nerocodes reviewed (approved):
LGTM

allebd committed:
feature(profile): create user profile

write using test to edit a user profile
create a route to edit a user profile
create user controller to edit a user profile

OvieMudi reviewed (approved):
LGTM
